src/navigation/menu_bar.py

```diff
@@ -362,10 +362,44 @@ def do_exit(self) -> None:
         """Exit app, with warning if current file unsaved"""
 
         async def coroutine(self: MenuNav) -> None:
+            def get_dynamic_filename(directory: str, generic: str) -> str:
```

Comment:
I recommend you use my implementation.

Reply:
Would've been better if this function is in `utils.py` since it doesn't use `self`